sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizedForm.cs

```diff
@@ -83,13 +85,21 @@ internal RecognizedForm(DocumentResult_internal documentResult, IReadOnlyList<Pa
             return fieldDictionary;
         }
 
-        private static IReadOnlyList<FormPage> ConvertPages(IReadOnlyList<PageResult_internal> pageResults, IReadOnlyList<ReadResult_internal> readResults)
+        private static IReadOnlyList<FormPage> ConvertUnsupervisedPages(PageResult_internal pageResult, IReadOnlyList<ReadResult_internal> readResults, int pageIndex)
+            => new List<FormPage> { new FormPage(pageResult, readResults, pageIndex) };
+
+        private IReadOnlyList<FormPage> ConvertSupervisedPages(IReadOnlyList<PageResult_internal> pageResults, IReadOnlyList<ReadResult_internal> readResults)
         {
             List<FormPage> pages = new List<FormPage>();
 
             for (int i = 0; i < readResults.Count; i++)
             {
-                pages.Add(new FormPage(pageResults != null ? pageResults[i] : null, readResults, i));
+                var pageNumber = readResults[i].Page;
+
+                if (pageNumber >= PageRange.FirstPageNumber && pageNumber <= PageRange.LastPageNumber)
+                {
+                    pages.Add(new FormPage(pageResults != null ? pageResults[i] : null, readResults, i));
+                }
```

Comment:
Checking range here so we only add the pages that are part of this form. I'm not doing `pageNumber = i + 1` because I don't think it's safe to assume we'll always get the pages back in order.

Reply:
Is it worth adding this as a comment as well?